myia/utils/misc.py

```diff
@@ -410,3 +410,8 @@ def __getitem__(self, name):
 def eprint(*things):
     """Print to stderr."""
     print(*things, file=stderr)
+
+
+def is_dataclass(cls):
+    """Returns whether cls is a dataclass."""
+    return isinstance(cls, type) and hasattr(cls, '__dataclass_fields__')
```

Comment:
This is super weird because the name doesn't imply that it checks for a type.  Perhaps improve the name?

Reply:
It does imply that though. A dataclass *is* a type.

Reply:
I expect that it checks for instances of dataclasses with this name.